wai-middleware-prometheus/src/Network/Wai/Middleware/Prometheus.hs

```diff
@@ -71,16 +79,33 @@ instrumentHandlerValue ::
      (Wai.Request -> Text) -- ^ The function used to derive the "handler" value in Prometheus
   -> Wai.Application -- ^ The app to instrument
   -> Wai.Application -- ^ The instrumented app
-instrumentHandlerValue = instrumentHandlerValueWithFilter Just
+instrumentHandlerValue = instrumentHandlerValueWithFilter Nothing Just
+
+-- | This function is used to populate the @handler@ label of all Prometheus metrics recorded by this library.
+--
+-- If you use this function you will likely want to override the default value
+-- of 'prometheusInstrumentApp' to be false so that your app does not get double
+-- instrumented.
+-- Added version label
+--
+-- WARNING: If you have 'ResponseRaw' values in your API, consider using
+-- @instrumentHandlerValueWithFilter ignoreRawResponses@ instead.
+instrumentHandlerValueWithVersionLabel ::
```

Comment:
Would it be more futuristic to take a (Text, Text) to define the key (as we have defaulted to "version" in this case) and value. 

We can even make it a [(Text, Text)] to be more appropriate.

Reply:
@aravindgopall yeah, I wanted to do like this first. But somehow when I use general label keys, not hardcoded, metrics worked not as expected. I had 5 identical lines with count 1:

```
http_request_duration_seconds_custom_count{handler="/ui/driver/location/",method="POST",status_code="200",version="DEV"} 1
http_request_duration_seconds_custom_count{handler="/ui/driver/location/",method="POST",status_code="200",version="DEV"} 1
http_request_duration_seconds_custom_count{handler="/ui/driver/location/",method="POST",status_code="200",version="DEV"} 1
http_request_duration_seconds_custom_count{handler="/ui/driver/location/",method="POST",status_code="200",version="DEV"} 1
http_request_duration_seconds_custom_count{handler="/ui/driver/location/",method="POST",status_code="200",version="DEV"} 1
```

instead of one line with count 5:

`http_request_duration_seconds_count{handler="/ui/driver/location/",method="POST",status_code="200",version="DEV"} 5`
So, I decided to use hardcoded label key `version`

Reply:
Sure. Let’s continue on this for now and have a issue created to discuss about the generalised labels approach.